inc/class-provider.php

```diff
@@ -90,13 +91,24 @@ protected function request( array $args ): MediaList {
 			) );
 		}
 
+		$total = absint( $response->get_headers()['x-wp-total'] );
+		$per_page = absint( $args['per_page'] ?? 40 );
+
 		if ( ! is_array( $response ) || ! $response ) {
-			return new MediaList();
+			return new MediaResponse(
+				new MediaList(),
+				$total,
```

Comment:
I feel this should be `0`.

```suggestion
				0,
```

If you agree, we might want to move the initialization of `$total` until after this condition.